File: app/src/main/java/com/akexorcist/localizationapp/nestedfragment/ParentFragment.java
package com.akexorcist.localizationapp.nestedfragment;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentTransaction;

import com.akexorcist.localizationapp.R;

/**
 * Created by Akexorcist on 7/22/15 AD.
 */
public class ParentFragment extends Fragment {
    public static Fragment newInstance() {
        return new ParentFragment();
    }

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        return inflater.inflate(R.layout.fragment_nested_parent, container, false);
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        if (savedInstanceState == null && getFragmentManager() != null) {
            Fragment fragment = ChildFragment.newInstance();
            FragmentTransaction transaction = getFragmentManager().beginTransaction();
            transaction.replace(R.id.layout_child_fragment_container, fragment);
            transaction.commit();
        }
    }
}

File: app/src/main/java/com/akexorcist/localizationapp/nestedfragment/ParentFragment.kt
package com.akexorcist.localizationapp.nestedfragment

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.fragment.app.commit
import com.akexorcist.localizationapp.databinding.FragmentNestedParentBinding

class ParentFragment : Fragment() {
    private lateinit var binding: FragmentNestedParentBinding

    companion object {
        fun newInstance(): Fragment = ParentFragment()
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
        binding = FragmentNestedParentBinding.inflate(layoutInflater, container, false)
        return binding.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        if (savedInstanceState == null) {
            parentFragmentManager.commit {
                replace(binding.layoutChildFragmentContainer.id, ChildFragment.newInstance())
            }
        }
    }
}
